### PR TITLE
More auto-margin fixes (some of them temporary)

### PR DESCRIPTION
**Pull request by etpinard**
This PR closes https://github.com/plotly/plotly.js/issues/4028 - which presents two ways of getting infinite redraw loops with `axis.automargin` turned on.

Example 1:
before: https://codepen.io/alanjmac/pen/oraKpR (try reducing the window width, squashing the graph)
after: https://codepen.io/etpinard/pen/OJLqVZK

Example 2:
before: https://codepen.io/wbrgss/pen/LYPryKO (open the console in Chrome, you should see `Uncaught RangeError: Maximum call stack size exceeded`)
after: https://codepen.io/etpinard/pen/xxKBGzW

Shoutout to @wbrgss for this example!

---

In brief, 

- https://github.com/plotly/plotly.js/commit/f506306b439db46e2f9de421aaddfb0d225ed988 is a small axis auto-margin fix
- https://github.com/plotly/plotly.js/commit/7a8eedf44b72456e8341802cc4efb7a19d6349fa essentially implements the
> Do not go back and forth between computed "auto" tick angle during the same Plots.doAutoMargin cycle

solution listed in https://github.com/plotly/plotly.js/issues/2704#issuecomment-524474154
- https://github.com/plotly/plotly.js/commit/631014a1688c4a495125c9774d8e8b438d995555 goes one step further to stay safe by capping the number of redraw calls `Plots.doAutoMargin` can trigger. This should remove all potential infinite redraw loops from our code. I'm hoping that this solution will be temporary and that by fixing our automargin pipeline problems (see https://github.com/plotly/plotly.js/issues/2704#issuecomment-534302295 for more on that), we'll be able to revert it.
- https://github.com/plotly/plotly.js/commit/58fd6327481a261fc2392725051bc9f81e8219b0 adds a mock that currently fails to render (as it leads to an infinite loop) and which also locks the https://github.com/plotly/plotly.js/commit/f506306b439db46e2f9de421aaddfb0d225ed988 fix 

----

@archmoj this one should go in `1.50.0` - looking for a quick review. Thanks!!

### CODE REVIEW

File: src/plots/cartesian/axes.js [hunk 1]
```diff
@@ -2584,19 +2589,36 @@ axes.drawLabels = function(gd, ax, opts) {
...
 
-    var done = Lib.syncOrAsync([allLabelsReady, fixLabelOverlaps]);
+    var seq = [allLabelsReady];
+
+    // N.B. during auto-margin redraw, if the axis fixed its label overlaps
+    // by rotating 90 degrees, do not attempt to re-fix its label overlaps
+    // as this can lead to infinite redraw loops!
+    if(fullLayout._redrawFromAutoMarginCount && prevAngle === 90) {
+        autoangle = 90;
+        seq.push(function() {
```
Comment (archmoj) [inline src/plots/cartesian/axes.js]: Do we want to name this new function?

**Comment by archmoj** [comment]
@etpinard 
:dancer: when last test passed on the CI.
Thanks.

**Comment by etpinard** [comment]
Thanks very much for the review @archmoj !

**Comment by fabiangehring** [comment]
Great thanks! Exactly what we are looking for. Any changes to get this released soon?

**Comment by etpinard** [comment]
> Any changes to get this released soon?

Next Monday. Sorry for the wait.

**Pull request merged**